contracts/impls/trustable/WitnetRequestBoardTrustableBase.sol

```diff
@@ -694,16 +694,18 @@ abstract contract WitnetRequestBoardTrustableBase
         // This would not be a valid encoding with CBOR and could trigger a reentrancy attack
         require(_cborBytes.length != 0, "WitnetRequestBoardTrustableDefault: result cannot be empty");
 
-        Witnet.Query storage _query = _state().queries[_queryId];
-        Witnet.Response storage _response = _query.response;
+        Witnet.Query storage __query = _state().queries[_queryId];
+        Witnet.Response storage __response = __query.response;
 
         // solhint-disable not-rely-on-time
-        _response.timestamp = _timestamp;
-        _response.drTxHash = _drTxHash;
-        _response.reporter = msg.sender;
-        _response.cborBytes = _cborBytes;
+        __response.timestamp = _timestamp;
+        __response.drTxHash = _drTxHash;
+        __response.reporter = msg.sender;
+        __response.cborBytes = _cborBytes;
 
-        _safeTransferTo(payable(msg.sender), _query.request.reward);
+        _safeTransferTo(payable(msg.sender), __query.request.reward);
         emit PostedResult(_queryId, msg.sender);
+
+        delete __query.request;
```

Comment:
This is the core change of this PR. Am I understanding correctly that from now, in every single case, the `request` part of queries will be dropped as soon as we have a result reported?

This is interesting. I can't really think of an use case at this point for archiving the requests. I'm very supportive of this idea.

Reply:
Great. However, some more changes to the data model will be proposed as to prevent this PR from becoming a breaking change (and therefore, forcing to either deploy new proxy address, or making 100% sure there are no pending requests in the WRB before upgrading it)

Reply:
That sounds like music to my ears! :heart_eyes:

Reply:
Done in https://github.com/witnet/witnet-solidity-bridge/pull/216/commits/f228c5953a91084d73c8e580ccae815f7741f9ad.